### ISSUE TITLE
Decide on client interface to find collections

### ISSUE THREAD

**Issue by lanthaler** [HydraCG/Heracles.ts#13]
This was discussed in #2 but merits an issue on its own. @elf-pavlik's current proposal is to implement something along the lines of 

```js
var operation = client.get("http://example.com")
    .getApiDocumentation()
    .getEntryPoint()
    .selectCollection({
      property: "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
      object: "http://schema.org/Event"
      })
    .getOperationOfType('http://schema.org/CreateAction');
```

with possible specialization such `selectCollectionByMembersType` as a for selecting a collection based on the type of its members:

```js
function selectCollectionByMembersType (membersType) {
  return this.selectCollection({
    property: "http://www.w3.org/1999/02/22-rdf-syntax-ns#type",
    object: membersType
  }))
}
```

**Comment by alien-mcl** [comment]
There is a PR #68 that adds support for these features. Example API:
```typescript
const apiDocumentation = await client.getApiDocumentation("http://example.com/api");
const entryPoint = await apiDocumentation.getEntryPoint();
const operation = entryPoint
    .collections.withMembersOfType("http://schema.org/Person").first()
    .operations.ofType("http://schema.org/CreateAction").first();
```

There are also two more methods:
- `withMembersMatching` - accepts property and object IRIs
- `withMembersInRelationWith` - accepts subject and property IRIs

**Comment by alien-mcl** [comment]
Implemented in PR #68
